frontend/src/Navigation.ts

```diff
@@ -17,14 +18,22 @@ export type GetAllExtensions = () => Promise<(HrefNavItem | NavSection)[]>;
 export type CalculateRoutes = (navIdentifier: [string, string], currentNamespace: string, extensions: (HrefNavItem | NavSection)[]) => RouteProps[];
 
 const getAllExtensions: GetAllExtensions = async () => {
-  return (
-    await Promise.all(
-      activePlugins.flatMap(async (pluginName: string) => {
-        const { extensions } = (await (await fetch(`/api/plugins/${pluginName}/plugin-manifest.json`))?.json()) || {};
-        return extensions;
-      }),
-    )
-  ).flat();
+  const extensions = (await Promise.allSettled(
+    activePlugins.flatMap(async ({ name: pluginName, pathPrefix = '/api/plugins' }: EnabledPlugin) => {
+      try {
+        const manifest = await (await fetch(`${pathPrefix}/${pluginName}/plugin-manifest.json`))?.json();
+        return manifest?.extensions;
+      } catch (e) {
+        console.log(e);
+      }
```

Comment:
I think we want it to explode... it'll reject the promise for us. 

Might want to do something like:
```
  results.filter(isRejectedPromise).forEach((e) => {
    // eslint-disable-next-line no-console
    console.error('Failed to fetch plugin-manifest', e.reason);
  });
```

Type-guard:
```
const isRejectedPromise = (result: PromiseSettledResult<Extension[]>): result is PromiseRejectedResult => {
  return result.status === 'rejected';
};
```

Reply:
Upon playing with it more, I went with this:

```
  const results: PromiseSettledResult<Extension[]>[] = await Promise.allSettled(
    activePlugins.flatMap(async ({ name: pluginName, pathPrefix = '/api/plugins' }: EnabledPlugin) => {
      const url = `${pathPrefix}/${pluginName}/plugin-manifest.json`;
      const response: Response = await fetch(url);
      if (response.status !== 200) {
        throw new Error(`${url} - ${response.status} - ${response.statusText}`);
      }
      const manifest: ConsolePluginManifestJSON = await response.json();
      return manifest.extensions;
    }),
  );
```
Let me know what you think.